src/Server/TCPHandler.cpp

```diff
@@ -820,8 +820,13 @@ void TCPHandler::runImpl()
 
                 if (exception_code == ErrorCodes::QUERY_WAS_CANCELLED_BY_CLIENT)
                     sendEndOfStream(query_state.value());
-                else
+                else if (exception->code() != ErrorCodes::SOCKET_TIMEOUT)
+                {
+                    /// The exception should not be send in case of timeout (SOCKET_TIMEOUT) during write, since:
+                    /// a) it will likely fail, and will lead to double timeout
+                    /// b) in case of it had been trown during writing Data packet, it may lead to CHECKSUM_DOESNT_MATCH on the receiver
                     sendException(*exception, send_exception_with_stack_trace);
+                }
```

Comment:
previously at line 819 we are also trying to send logs - don't you think we should avoid it either in case of timeout?

Reply:
Indeed, thanks, fixed.